.github/actions/Build_and_Test_CppInterOp/action.yml

```
@@ -58,7 +64,10 @@ runs:
           cmake --build . --target check-cppinterop --parallel ${{ env.ncpus }}
           os="${{ matrix.os }}"
           if [[ "${os}" != "macos"* ]]; then
-            valgrind --show-error-list=yes --track-origins=yes --error-exitcode=1 unittests/CppInterOp/CppInterOpTests/unittests/bin/${{ env.BUILD_TYPE }}/CppInterOpTests
+            valgrind --leak-check=full --show-leak-kinds=all --show-error-list=yes --track-origins=yes --error-exitcode=1 unittests/CppInterOp/CppInterOpTests/unittests/bin/${{ env.BUILD_TYPE }}/CppInterOpTests
+          fi
+          if [[ "${{ matrix.oop-jit }}" == "On" ]]; then
+            ./unittests/CppInterOp/CppInterOpTests/unittests/bin/${{ env.BUILD_TYPE }}/CppInterOpTests --use-oop-jit
```

```suggestion
          if [[ "${{ matrix.oop-jit }}" == "On" ]]; then
            ./unittests/CppInterOp/CppInterOpTests/unittests/bin/${{ env.BUILD_TYPE }}/CppInterOpTests --use-oop-jit
            valgrind --leak-check=full --show-leak-kinds=all --show-error-list=yes --track-origins=yes --error-exitcode=1 ./unittests/CppInterOp/CppInterOpTests/unittests/bin/${{ env.BUILD_TYPE }}/CppInterOpTests --use-oop-jit
```

We first want to run the OOP-JIT tests without valgrind, then with valgrind.